### DISCUSSION TITLE
Unable to build react application with the new version 0.0.17

### DISCUSSION THREAD

**Discussion by shuvradeb-saha** [dlemstra/magick-wasm#64] [Q&A]
I had been using the magick-wasm v0.0.15 in a react application which was going well. But after upgrading the library to 0.0.17, I started to get the following error during `yarn build`:-
```
RangeError: /react-app/node_modules/@imagemagick/magick-wasm/dist/index.mjs: Maximum call stack size exceeded
    at RegExp.exec (<anonymous>)
    at tokenize.next (<anonymous>)
```
I have been using the following node environment:
`Node: v16.14.2`
`Yarn: 1.22.19`

The changes are made only in the import statement section of my own code after the upgrade.

For the previous `magick-wasm v0.0.15` version we had to import like this:
```
import { ImageMagick, initializeImageMagick } from '@imagemagick/magick-wasm';
import { Percentage } from '@imagemagick/magick-wasm/percentage';
```
But in the latest version (`v0.0.17`) importing everything from a single import statement like this:
```
import { ImageMagick, initializeImageMagick, Percentage } from '@imagemagick/magick-wasm';
```

I tried clearing the cache and re-installing the node modules inside the applications, but nothing worked. Any help regarding this would be appreciated.

**Comment by dlemstra** [comment]
The latest version use different distribution method that was added by @Peeterush. One of these changes was a different import method. I have heard different points of view about this and I wonder if I should revisit that strategy.

It might be possible that your code had a bug that you never noticed but is now visible due to these changes. Is it possible to create a small application in a GitHub repository that can be used to reproduce this issue.
